src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -338,7 +344,7 @@ private static Task CompleteSwitchingBrokerAsync(IRSessionSwitchBrokerTransactio
             using (await _connectArwl.ReaderLockAsync(ct)) {
                 try {
                     var hostLoad = await Broker.GetHostInformationAsync<HostLoad>(ct);
-                    OnBrokerStateChanged(hostLoad);
+                    OnHostLoadChanged(hostLoad);
```

Comment:
That is too much of a split. If connection dies, we won't be notified about it anymore.
I'd rather leave one method, call it something like `OnBrokerStateOrHostLoadChanged`, but raise `BrokerStateChanged` only if `IsConnected` has changed, while `HostLoadChanged` would've been raised all the time.